### PR TITLE
Call and unary precedence

### PR DESCRIPTION
**Pull request by doras100p**
Fix the precedence of call expressions and unary expression, and add a corresponding test

### CODE REVIEW

**Comment by doras100p** [comment]
I'm not sure if this is the right way to fix this problem, but hopefully if not it would encourage someone to fix it differently :)
The test describes the problem best.

**Comment by doras100p** [comment]
The problem also exists for call and binary, the fix is the same, I added a test to that as well.

File: grammar.js [hunk 1]
```diff
@@ -709,7 +710,7 @@ module.exports = grammar({
         ),
 
         // Expressions
-        _expression: $ => choice(
+        _expression: $ => prec.right(choice(
```
Comment (JoranHonig) [inline grammar.js]:
```suggestion
        _expression: $ => choice(
```

I think it is better if the associativity rule is only applied to calls

File: grammar.js [hunk 2]
```diff
@@ -860,10 +861,10 @@ module.exports = grammar({
             field('right', $._expression)
         )),
 
-        call_expression: $ => seq(
+        call_expression: $ => prec(PREC.CALL, seq(
```
Comment (JoranHonig) [inline grammar.js]:
```suggestion
        call_expression: $ => prec.right(PREC.CALL, seq(
```

File: grammar.js [hunk 3]
```diff
@@ -721,7 +722,7 @@ module.exports = grammar({
             $.struct_expression,
             $.ternary_expression,
             $.type_cast_expression,
-        ),
+        )),
```
Comment (JoranHonig) [inline grammar.js]:
```suggestion
        ),
```

**Review by JoranHonig — COMMENTED** [review]
Thanks for the contribution! 

I applied some changes isolating the associativity rule to the call function.

**Comment by doras100p** [comment]
@JoranHonig I think something is still broken, I need to reproduce it with a test or something, but I still sometimes get !a() as a call to (!a)

**Pull request merged**